drivers/wifi/siwx917/siwx917_wifi.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * The licensor of this software is Silicon Laboratories Inc. Your use of this
```

Comment:
Is there a reason to not use Apache-2.0 for this?

Reply:
> Is there a reason to not use Apache-2.0 for this?

That's indeed a fair question, and if it's not possible to use anything else (at least in the short term), seems like `hal_silabs` may be a better place for it? It'd be good to keep the code in `zephyr-silabs` upstreamable at least from a license perspective.

Reply:
Since this is a Zephyr implementation that merely _uses_ the Wiseconnect API, I can't think of anything that should block this from having a Zephyr friendly license, even if Wiseconnect iself hasn't been relicensed yet.